File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_table_ids.c


#include "ses_defines.h"
#include "ses_globals.h"
#include "ses_internals.h"
#include "ses_externs.h"

#include "stdio.h"
#include <stdlib.h>

#include <stdio.h>

#undef DEBUG_WRAP

#ifdef LC_UNDER
void ses_get_table_ids_(ses_file_handle* pt_handle, ses_material_id* mid, long* the_size, ses_table_id_reference the_return) {
#else

#ifdef UC_UNDER
void SES_get_table_ids_(ses_file_handle* pt_handle, ses_material_id* mid, long* the_size, ses_table_id_reference the_return) {
#else

#ifdef LC_NOUNDER
void ses_get_table_ids(ses_file_handle* pt_handle, ses_material_id* mid, long* the_size, ses_table_id_reference the_return) {
#else

#ifdef UC_NOUNDER
void SES_get_table_ids(ses_file_handle* pt_handle, ses_material_id* mid, long* the_size, ses_table_id_reference the_return) {
#endif

#endif
#endif
#endif

  ses_table_id_reference return_value = (ses_table_id_reference)NULL;

  ses_file_handle the_handle = *pt_handle;
  ses_material_id the_mid = *mid;
#ifdef DEBUG_WRAP
  printf("wrap_ses_get_table_ids.c:  the_handle is %d\n", the_handle);
#endif

  return_value = ses_get_table_ids(the_handle, the_mid, the_size);
  if (return_value != NULL) {
#ifdef DEBUG_WRAP 
    printf("wrap_ses_get_table_ids.c:  return_value[0] is %d\n", return_value[0]);
    printf("wrap_ses_get_table_ids.c:  the_size is %d\n", the_size[0]);
#endif

    int i = 0;
    for (i=0; i < the_size[0]; i++) {
      the_return[i] = return_value[i];
    }
 }
 else {
    the_return = NULL;
 }
}

 

File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_table_ids.f90

#undef DEBUG_32BIT

#undef DEBUG_WRAP



function ses_get_table_ids_f90(the_handle, mid, the_size)

   implicit none

   SES_FILE_HANDLE, INTENT(IN) :: the_handle
   SES_MATERIAL_ID :: mid
   LONG_POINTER, dimension(:) :: the_size
 
   SES_TABLE_ID_REFERENCE, dimension(:) :: ses_get_table_ids_f90
   SES_TABLE_ID, dimension(1000), target :: actual_table

   SES_TABLE_ID :: ses_get_table_ids
#ifdef DEBUG_32BIT
   integer(kind = 4), dimension(1000), target :: actual_table_4
   integer(kind = 8), dimension(1000), target :: actual_table_8
   integer(kind = 4), dimension(:), pointer :: return_4
   integer(kind = 8), dimension(:), pointer :: return_8
#endif

   SES_TABLE_ID, target :: rv_target(2)
   integer :: dummy

   ses_get_table_ids_f90=>actual_table
#ifdef DEBUG_32BIT
   return_4=>actual_table_4
   return_8=>actual_table_8
#endif
 
#ifdef DEBUG_WRAP
   print *, "ses_get_table_ids_f90:  the_handle is ", the_handle
   print *, "ses_get_table_ids_f90:  mid is ", mid
#endif

#ifdef DEBUG_32BIT
   dummy = ses_get_table_ids(the_handle, mid, the_size, return_4)
   ses_get_table_ids_f90 = return_4
#else
   dummy = ses_get_table_ids(the_handle, mid, the_size, ses_get_table_ids_f90)
#endif

#ifdef DEBUG_WRAP
   print *, "ses_get_table_ids_f90:  ses_get_table_ids_f90 is ", ses_get_table_ids_f90(1)
   print *, "ses_get_table_ids_f90:  the_size is ", the_size(1)
#endif

 
end function ses_get_table_ids_f90


